tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -709,6 +709,47 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system",
 	},
 
+	/** Firmware layout for the RE350 */
+	{
+		.id = "RE350",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:RE350,product_ver:1.0.0,special_id:45550000}\n"
+			"{product_name:RE350,product_ver:1.0.0,special_id:00000000}\n"
+			"{product_name:RE350,product_ver:1.0.0,special_id:41550000}\n"
+			"{product_name:RE350,product_ver:1.0.0,special_id:55530000}\n"
+			"{product_name:RE350,product_ver:1.0.0,special_id:43410000}\n"
+			"{product_name:RE350,product_ver:1.0.0,special_id:4b520000}\n"
+			"{product_name:RE350,product_ver:1.0.0,special_id:4a500000}\n",
+		.support_trail = '\x00',
+		.soft_ver = NULL,
+
+		/**
+			The original os-image partition is too small,
+			so we enlarge it to 1.5M, this is just large enough
+		*/
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x20000},
+			{"os-image", 0x20000, 0x180000},
+			{"file-system", 0x200000, 0x400000},
```

Comment:
Wrong calculate. os-image end at 0x1A0000, file-system starts at 0x200000. You waste 512KB